src/connection/query.ts

```diff
@@ -84,20 +114,33 @@ export class Query<T> {
     let result = await this.job.send(JSON.stringify(queryObject));
 
     let queryResult: QueryResult<T> = JSON.parse(result);
-    this.state = queryResult.is_done? QueryState.RUN_DONE: QueryState.RUN_MORE_DATA_AVAILABLE;
+    this.state = queryResult.is_done ? QueryState.RUN_DONE : QueryState.RUN_MORE_DATA_AVAILABLE;
 
     if (queryResult.success !== true) {
       this.state = QueryState.ERROR;
       throw new Error(queryResult.error || `Failed to run query (unknown error)`);
     }
     return queryResult;
   }
+
   public async close() {
     this.state = QueryState.RUN_DONE;
+
+    if (this.correlationId) {

```

Comment:
should be 
```typescript
if (this.correlationId && this.state !== QueryState.RUN_DONE)
```
and of course moved above the preceding assignment

Reply:
Are you suggesting that `RUN_DONE` indicates that the cursor is closed?

Reply:
yes. If it's `RUN_DONE` the cursor is closed

Reply:
future PR is improved enum values :eyes: